platform/bukkit/src/main/java/me/lokka30/treasury/plugin/bukkit/hooks/papi/economy/BalanceCache.java

```diff
@@ -104,52 +102,47 @@ private void proceed(
         }
         OfflinePlayer player = players.get(currentIndex);
         provider.hasAccount(AccountData.forPlayerAccount(player.getUniqueId())).thenCompose(res -> {
-            if (!res.isSuccessful()) {
-                return CompletableFuture.completedFuture(Response.failure(res.getFailureReason()));
-            }
-            if (res.getResult()) {
+            if (res) {
                 return provider.accountAccessor().player().withUniqueId(player.getUniqueId()).get();
             }
-            return CompletableFuture.completedFuture(Response.failure(FailureReason.of(
-                    "accountNotExists")));
-        }).whenComplete((res, ex) -> {
+            return FutureHelper.failedFuture(new TreasuryException("accountNotExists"));
+        }).whenComplete((account, ex) -> {
             if (ex != null) {
-                throw new RuntimeException("An error occurred whilst updating balance cache", ex);
-            }
-
-            if (!res.isSuccessful()) {
-                if (!res.getFailureReason().getDescription().equalsIgnoreCase("accountNotExists")) {
-                    // log the problem and proceed with next entry
-                    TreasuryPlugin.getInstance().logger().error(
-                            "Error whilst trying to update balance cache for " + (player.getName() != null
-                                    ? player.getName()
-                                    : player.getUniqueId().toString()) + ": " + res
-                                    .getFailureReason()
-                                    .getDescription());
+                if (ex instanceof TreasuryException) {
+                    if (!ex.getMessage().equalsIgnoreCase("accountNotExists")) {
```

Comment:
`"accountNotExists"` should be a constant somewhere.

Reply:
see above; it's used in `.thenCompose`. also this is not API